platform/security/common/src/main/java/org/codice/ddf/security/common/Security.java

```diff
@@ -291,14 +297,21 @@ public SecurityManager getSecurityManager() {
     }
 
     public static <T> T runAsAdmin(PrivilegedAction<T> action) {
-        Set<Principal> principals = new HashSet<>();
-        principals.add(new RolePrincipal("admin"));
-        javax.security.auth.Subject subject = new javax.security.auth.Subject(true,
-                principals,
-                new HashSet(),
-                new HashSet());
+        return javax.security.auth.Subject.doAs(JAVA_ADMIN_SUBJECT, action);
+    }
 
-        return javax.security.auth.Subject.doAs(subject, action);
+    public static <T> T runAsAdminPea(PrivilegedExceptionAction<T> action)
```

Comment:
This is a *really* weird name for a method. When I saw it referenced above, I assumed it was a typo in the code. Why not just overload `runAsAdmin`? `PrivilegedAction` and `PrivilegedExceptionAction` are neither assignable from the other so you'll have no type confusion. If you must have a different name, than I'd suggest something like `runAsAdminWithException`. I don't love that, but it's clearer.